### PR TITLE
Enables JPS pathfinding to path through z levels

### PR DESCRIPTION
**Pull request by TheDarkElites**
## About the Pull Request

JPS pathfinding can now handle different Z-Levels.

Also some minor fixes to 173's AI to take advantage of z-level pathfinding.

## Why It's Good For The Game

Better pathfinding = better AI for SCPs and other NPCs.

## Changelog

:cl:
add: Z-Level Pathfinding
add: New pathfinder debug verb to help debug pathfinding.
fix: 173 now properly stalks individuals through z-levels."
fix: 173 AI should no longer be able to kill when looked at in certain situations.
balance: 173 now stalks targets rather than losing interest once it can no longer see them. Min distance to lose 173 is 14.
/:cl:

### CODE REVIEW

File: code/_helpers/path.dm
Comment (Tsurupeta) [inline code/_helpers/path.dm]: what about start == end?

Comment (TheDarkElites) [reply code/_helpers/path.dm]: will add back

**Comment by TheDarkElites** [comment]
Dont merge I just realized I was doing something redundant

**Comment by TheDarkElites** [comment]
now once again ready for review/merge

**Review by Tsurupeta — DISMISSED** [review]

**Review by Tsurupeta — DISMISSED** [review]
yolo

**Review by Tsurupeta — APPROVED** [review]

**Pull request merged**
